flytekit/core/type_engine.py

```diff
@@ -1642,12 +1642,16 @@ def get_dict_types(t: Optional[Type[dict]]) -> typing.Tuple[Optional[type], Opti
         _origin = get_origin(t)
         _args = get_args(t)
         if _origin is not None:
-            if _origin is Annotated:
-                raise ValueError(
-                    f"Flytekit does not currently have support \
-                        for FlyteAnnotations applied to dicts. {t} cannot be \
-                        parsed."
-                )
+            if _origin is Annotated and _args:
+                # _args holds the type arguments to the dictionary, i.e. get_args(Dict[int, str]) == (int, str).
```

Comment:
Although this comment is informational, I find a little out of place and can be confusing to a reader. Concretely, inside this `if` block `_origin` is always `Annotated` and not `Dict`.